### PR TITLE
Table: Add SortableHeaderCell End align

### PR DESCRIPTION
**Pull request by rlingineni**
### Summary

Enabling the SortableHeaderCell to align to the right as expected. Previously, the arrow was in the way for 

#### What changed?

Adding an `align= start | end` and moving the icon when end-aligned. 

The end property also reverses the order so that the text is the last thing to appear.

#### Why?

We have a few [customers](https://jira.pinadmin.com/secure/RapidBoard.jspa?rapidView=1936&view=planning&selectedIssue=GESTALT-5154&quickFilter=12031&issueLimit=100) that ran into issues with this.

Before:
The icon would go away, and the alignment looks poor

image

After:

LTR
image

RTL
image

### Links

- [Jira](https://jira.pinadmin.com/secure/RapidBoard.jspa?rapidView=1936&view=planning&selectedIssue=GESTALT-5154&quickFilter=12031&issueLimit=100)
- [Figma](https://www.figma.com/file/HbBD5AgCNvupMVKBkTgdsH/Table-Enhancements?type=design&node-id=603-45434&mode=design&t=XS678P7HLRVDvyCg-0)

### Checklist

- [X] Added unit and Flow Tests
- [X] Added documentation + accessibility tests
- [X] Verified accessibility: keyboard & screen reader interaction
- [X] Checked dark mode, responsiveness, and right-to-left support
- [X] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

**Comment by AlbertCarreras** [comment]
@rlingineni  Can we add this to TypeScript?

**Comment by AlbertCarreras** [comment]
@rlingineni  Can we add this to TypeScript?

**Review by AlbertCarreras — APPROVED** [review]

**Comment by AlbertCarreras** [comment]
TypeScript changes, please.

File: packages/gestalt/src/TableSortableHeaderCell.js [hunk 1]
```diff
@@ -97,10 +107,17 @@ export default function TableSortableHeaderCell({
...
+            alignItems="center"
+            justifyContent={align}
+            dangerouslySetInlineStyle={{
+              __style: { flexDirection: align === 'end' ? 'row-reverse' : 'row' },
+            }}
+          >
             {children}
             <Box
-              marginStart={2}
+              {...iconMarginProps}
```
Comment (AlbertCarreras) [inline packages/gestalt/src/TableSortableHeaderCell.js:120]: don't spread,  future codemode issues

Comment (AlbertCarreras) [reply packages/gestalt/src/TableSortableHeaderCell.js:120]: inline

**Comment by AlbertCarreras** [comment]
The changes overall look good to me

Pleae, fix 2 comments

**Review by AlbertCarreras — APPROVED** [review]
LGTM but fix 2 comments

File: packages/gestalt/src/TableSortableHeaderCell.js (hunk 1 above)
Comment (rlingineni) [reply packages/gestalt/src/TableSortableHeaderCell.js:120]: Oh that's right! Okay, will use inline

**Pull request merged**